src/products/views/ProductUpdate/handlers/utils.ts

```diff
@@ -150,3 +132,136 @@ export function getProductChannelsUpdateVariables(
     },
   };
 }
+
+export function getCreateVariantInput(data: DatagridChangeOpts, index: number) {
+  return {
+    attributes: getAttributeData(data.updates, index, data.removed),
+    sku: getSkuData(data.updates, index, data.removed),
+    name: getNameData(data.updates, index, data.removed),
+    channelListings: getVariantChannelsInputs(data, index),
+    stocks: getStockData(data.updates, index, data.removed),
+  };
+}
+
+export function getBulkVariantUpdateInputs(
+  variants: ProductFragment["variants"],
+  data: DatagridChangeOpts,
+): ProductVariantBulkUpdateInput[] {
+  return variants
+    .map(
+      (variant, variantIndex): ProductVariantBulkUpdateInput => ({
+        id: variant.id,
+        attributes: getAttributeData(data.updates, variantIndex, data.removed),
+        sku: getSkuData(data.updates, variantIndex, data.removed),
+        name: getNameData(data.updates, variantIndex, data.removed),
+        stocks: getStockData(data.updates, variantIndex, data.removed),
+        channelListings: getVariantChannelsInputs(data, variantIndex),
+      }),
+    )
+    .filter(
+      variant =>
+        variant.name ||
+        variant.sku ||
+        variant.attributes.length > 0 ||
+        variant.stocks.length > 0 ||
+        variant.channelListings.length > 0,
+    );
+}
```

Comment:
sugestion: for chaing calls like this you can benefit from function composing eg like this:

```
const toUpdateInput = (variant, variantIndex): ProductVariantBulkUpdateInput => ({
  id: variant.id,
  attributes: getAttributeData(data.updates, variantIndex, data.removed),
  sku: getSkuData(data.updates, variantIndex, data.removed),
  name: getNameData(data.updates, variantIndex, data.removed),
  stocks: getStockData(data.updates, variantIndex, data.removed),
  channelListings: getVariantChannelsInputs(data, variantIndex),
})

const byAvailability = variant =>
  variant.name ||
  variant.sku ||
  variant.attributes.length > 0 ||
  variant.stocks.length > 0 ||
  variant.channelListings.length > 0

export function getBulkVariantUpdateInputs(
  variants: ProductFragment["variants"],
  data: DatagridChangeOpts,
): ProductVariantBulkUpdateInput[] {
  return variants
    .map(toUpdateInput)
    .filter(byAvailability);
}
```

Please notice that right now, when i read entire chain I can almost use natural language. Also each function has one resposibility